ludwig/api.py

```diff
@@ -104,6 +106,88 @@
 logger = logging.getLogger(__name__)
 
 
+@dataclass
```

Comment:
Should we maybe move these dataclasses into an auxilary file and import them to maintain the api file flow? I feel like these data classes could be pretty easily compartmentalized into a dedicated file which may help with the mental model associated with the api file. WDYT?

Reply:
I put them here because they are used as data types in method of api.py, and I couldn't think of a better idea.

On that note, Arnav and I have discussed further refactoring of api.py this sprint, perhaps defining LudwigModel in `ludwig_model.py` and pulling it into api.py.  So maybe we can separate that out into a follow-up PR.